featuretools/primitives/utils.py

```diff
@@ -227,3 +227,46 @@ def _find_class_in_descendants(self, search_key):
 
             if cls_key == search_key:
                 return cls
+
+
+def _roll_series_with_gap(series, window_size, gap=0, min_periods=1):
+    """Provide rolling window calculations with window specified with both a window length
+        and a gap that indicates the amount of time between each instance and its window.
+
+    Args:
+        series (Series): The series over which the windows will be created. Must be numeric in nature
+            and have a datetime64[ns] index.
+        window_length (int): The number of rows to be included in each window. For data
+            with a uniform sampling frequency, for example of one day, the window_length will
+            correspond to a period of time, in this case, 7 days for a window_length of 7.
+        gap (int, optional): The number of rows prior to each instance to be skipped before
```

Comment:
nit-pick: The use of "skipped" confused me at first. If gap=1, and we're saying that 1 row prior to each instance will be skipped, that makes me think it's doing what gap=2 actually does.